rstcheck/__init__.py

```diff
@@ -517,17 +523,41 @@ def load_configuration_from_file(directory: str, args: argparse.Namespace) -> ar
 
 def _get_options(directory_or_file: str, debug: bool = False) -> typing.Dict[str, str]:
     config_path = find_config(directory_or_file, debug=debug)
-    if not config_path:
+    if not config_path:  # pylint: disable=no-else-return
         return {}
+    else:
+        config_file = os.path.basename(config_path)
 
-    parser = configparser.ConfigParser()
-    parser.read(config_path)
-    try:
-        options = dict(parser.items("rstcheck"))
-    except configparser.NoSectionError:
-        return {}
+    if config_file == "pyproject.toml" and TOMLI_INSTALLED:  # pylint: disable=no-else-return
```

Comment:
When I made these changes, I also had to change the default return from ```find_config``` to an empty string rather than ```None``` and change the return type from ```Optional[str]``` to ```str```.  Everything else I tried resulted in either mypy or pylint errors.

I also wonder if the except block should print a message letting the user know an rstcheck section wasn't found in the config file?  This would require something similar in the ```_get_pyproject_options``` function too, but I could see it saving many late night hours of frustration because someone misspelled rstcheck in their config file.  I'm also OK raising an issue requesting this if you want to prevent scope creep in this PR.

Reply:
> I also wonder if the except block should print a message letting the user know an rstcheck section wasn't found in the config file?

Only in debug mode, I think this would be plausible I guess. Or one could argue to print the warning if the pyproject.toml file is set via --config. But then this should also be done to setup.cfg. I think this is better done/discussed in another PR/issue.

Reply:
Raised issue #99 for further discussion.